src/scripts/matcher.js

```diff
@@ -115,8 +112,8 @@ function handleNoMatch(url) {
     matchedPattern: NOMATCH_TEXT,
     timestamp: Date.now()
   });  
-  chrome.browserAction.setIcon(NOMATCH_ICON);
-  chrome.browserAction.setTitle(NOMATCH_TITLE);
-  chrome.browserAction.setBadgeText(NOMATCH_BADGE_TEXT); 
-  browser.browserAction.setBadgeBackgroundColor(NOMATCH_BACKGROUND_COLOR);  
+  chrome.browserAction.setIcon({path: 'images/gray.svg'});
+  chrome.browserAction.setTitle({title: NOMATCH_TEXT});
+  chrome.browserAction.setBadgeText({text: NONE_TEXT}); 
+  browser.browserAction.setBadgeBackgroundColor({color: NOMATCH_COLOR});  
 }
```

Comment:
But this creates new objects for the arguments to setIcon(), setTitle(), setBadgeText(), and setBadgeBackgroundColor() every single time handleNoMatch() is called. Those object creations must allocate memory and they also need to be garbage collected. I created the constants to avoid that. Does that make sense?

Reply:
This will be in the next commit per our discussion. Current PR conflicts with HEAD. Thanks for your contribution and comments in chat; they are greatly appreciated.